src/main/system-analysis/service/error.service.ts

```
@@ -0,0 +1,265 @@
+    resetTimeout: 30000, // 30 seconds
+    monitoringPeriod: 60000, // 1 minute
+  };
+
+  return new CircuitBreaker({ ...defaultOptions, ...options });
+};
+
+/**
+ * Module-level cache for circuit breaker instances
+ * Uses WeakMap to allow garbage collection of unused function references
+ */
+const circuitBreakerCache = new WeakMap<Function, CircuitBreaker>();
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)